### ISSUE TITLE
Omit precise cast when saving Serializable

### ISSUE THREAD

**Issue by AlexShafir** [evernote/android-state#44]
First I would like to say many thanks for `@StateReflection` as it allows me to have private fields!

Now I see that when saving `Serializable` object precise cast is used when cast to `Serializable` would suffice.

This prevents, for instance, having private enums as type for fields.

Library Version 1.2.0

Generated code:
```java
public class MyMapPresenter$$StateSaver<T extends MyMapPresenter> extends Injector.Object<T> {
  private static final HashMap<String, Bundler<?>> BUNDLERS = new HashMap<String, Bundler<?>>();

  private static final InjectionHelper HELPER = new InjectionHelper("some.very.long.path.MyMapPresenter$$StateSaver", BUNDLERS);

  static {
  }

  @Override
  @SuppressWarnings("unchecked")
  public void save(T target, Bundle state) {
    try {
      Field field = target.getClass().getDeclaredField("state");
      boolean accessible = field.isAccessible();
      if (!accessible) {
        field.setAccessible(true);
      }
      // <<<<<<<<<<<<<<<< Just cast to Serializable >>>>>>>>>>>>>>>>>>>>>>>>
      HELPER.putSerializable(state, "state", (some.very.long.path.MyMapPresenter.State) field.get(target));
      if (!accessible) {
        field.setAccessible(false);
      }
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }

  @Override
  @SuppressWarnings("unchecked")
  public void restore(T target, Bundle state) {
    try {
      Field field = target.getClass().getDeclaredField("state");
      boolean accessible = field.isAccessible();
      if (!accessible) {
        field.setAccessible(true);
      }
      field.set(target, HELPER.getSerializable(state, "state"));
      if (!accessible) {
        field.setAccessible(false);
      }
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }
}

```

**Comment by vRallev** [comment]
I don't understand why this is a problem. I just tested it and don't run into issues. Can you provide the sample class?

**Comment by AlexShafir** [comment]
Here is an example:

```java
public class MyMapPresenter {
    // <<<<<< Try setting this enum as private and you will get Compile-Time Error  >>>>>>
    enum State {STARTED, STOPPED, PAUSED}

    @StateReflection
    private State state;

    private MyMapView view;
}
```

**Comment by vRallev** [comment]
Oh, yes, that's expected. But this will also happen with any other private class. That's not a limitation of the library, that's simply how the language scopes work.

**Comment by AlexShafir** [comment]
If library omits direct cast and instead makes cast to Serializable, then it will work.

Imagine that I don't use your library, will I make my enum private or non-private? Surely private.

If you prefer no-fix it, could you point out place in code where cast happens?
I think one should do a fork.

**Comment by vRallev** [comment]
I'll take a look

**Comment by AlexShafir** [comment]
Any traction here?

**Comment by vRallev** [comment]
This is fixed in version `1.3.0`
